src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -445,6 +451,24 @@ private IDataView GetShuffledData(string path)
                         statisticsCallback(metrics);
                     }
                 }
+                // early stopping check
+                if (options.EnableEarlyStopping)
+                {
+                    if (metrics.Train.Accuracy - options.MinDelta > history.Accuracy)
+                    {
+                        history.Accuracy = metrics.Train.Accuracy;
+                        wait = 0;
+                    }
+                    else
+                    {
+                        wait += 1;
+                        if (wait >= options.Patience)
+                        {
+                            Console.WriteLine("*** Early Stopping at epoch " + epoch.ToString());
+                            earlyStop = true;
+                        }
+                    }
```

Comment:
Please do not put a PR without a unit-test, if you do, please mark it as Draft PR or WIP.